test/stan_connection_lost.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2016-2018 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* jslint node: true */
+/* global describe: false, beforeEach: false, afterEach: false, it: false */
+'use strict';
+
+var STAN = require ('../lib/stan.js'),
+NATS = require('nats'),
+nuid = require('nuid'),
+ssc = require('./support/stan_server_control'),
+should = require('should'),
+timers = require('timers');
+
+describe('Stan Server Ping Specific', function() {
+
+  var PORT = 9876;
+  var cluster = 'test-cluster';
+  var uri = 'nats://localhost:' + PORT;
+  var server = null;
+
+  beforeEach(function (done) {
+    server = ssc.start_server(PORT, function () {
```

Comment:
These tests are fine for nats-hosted streaming server - but the more likely condition is that the gnatsd is running standalone. In those cases the tests below which test for disconnect will fail. The tests startup should do something like this:
```beforeEach(function (done) {
    nats_server = ssc.start_server(NATS_PORT, ['-cid', 'not-test-cluster'], function() {
      stan_server = ssc.start_server(PORT, ['-ns', 'nats://localhost:' + NATS_PORT], function () {
        timers.setTimeout(function () {
          done();
        }, 250);
      });
    });
  });```